src/API.jl

```diff
@@ -1141,7 +1144,8 @@ function precompile(ctx::Context; internal_call::Bool=false, strict::Bool=false,
                     end
                     try
                         ret = Logging.with_logger(Logging.NullLogger()) do
-                            Base.compilecache(pkg, sourcepath, iob, devnull) # capture stderr, send stdout to devnull
+                            # capture stderr, send stdout to devnull, don't skip loaded modules
+                            Base.compilecache(pkg, sourcepath, iob, devnull, false)
```

Comment:
@IanButterworth is this change not backportable to 1.6.2?

Reply:
https://github.com/JuliaLang/julia/pull/40345 needs backporting to the base release branch, then CI here needs to be run against that branch

Reply:
Could https://github.com/JuliaLang/julia/pull/40702 be merged early and another started to continue base backports, so that this can be tested here?

Reply:
https://github.com/JuliaLang/julia/pull/40702#issuecomment-854675822